### PR TITLE
A user can place a referral and is taken to the Find a Bedspace journey and it remembers the referral data (part 1)

### PR DESCRIPTION
**Pull request by bryangaledxw**
# Changes in this PR

This PR is the first step of having a banner at the top of each page that is part of the journey from a referral that is ready to place, through to successfully booking a bedspace.

This PR adds a banner to the top of the bedspace search page indicating the context, and prefills this page's start date fields. A future PR will build on this, and preserve the banner through until a bedspace is successfully booked.

As development of this feature as a whole has involved a tour across the codebase, this PR begin with a lot of minor tidying, followed by building up the utility functions we will use in this feature, then the nunjucks macros, then finally putting them to use, and integration this initial stub of the feature.

This PR takes takes the approach of tracking place context with a query parameter in a URL. Though this approach was initially chosen for expediency over a session-based approach, it has the advantages that what happens in one tab will not influence what happens in another tab, and if the context is "cleared" it is retrievable using the back button

### Before
localhost_3000_find-a-bedspace

### After
localhost_3000_find-a-bedspace_placeContextAssessmentId=caa81af5-0f3c-45d6-986b-c007fdf8007f (1)

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by bryangaledxw** [comment]
[A user can place a referral and is taken to the Find a Bedspace journey and it remembers the referral data (part 1)](https://trello.com/c/mbxrCkYt/1488-a-user-can-place-a-referral-and-is-taken-to-the-find-a-bedspace-journey-and-it-remembers-the-referral-data-part-1)

File: cypress_shared/pages/assess/summary.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class AssessmentSummaryPage extends Page {
 
   clickAction(option: string) {
     cy.get('.moj-button-menu__toggle-button').contains('Update referral status').click()
-    cy.get('.moj-button-menu__wrapper').contains(option).click()
+    cy.get('.moj-button-menu__wrapper').contains(option).invoke('removeAttr', 'target').click()
```
Comment (libuk) [inline cypress_shared/pages/assess/summary.ts:22]: Elegant workaround.

**Review by libuk — APPROVED** [review]
Nicely paced and laid out. The changes here make sense, and I couldn't find any obvious flaws in the approach.

**Comment by bryangaledxw** [comment]
@selina-dxw @bazzle Will merge this for the sake of unblocking things - can roll any design changes into the next PR

**Comment by selina-dxw** [comment]
Thanks for this, just a couple things:

Is it possible to pull through the person's gender rather than sex? 

In the designs we had 'Suitable to share' instead of 'Accommodation required from'. Is that something that will come in a later iteration? 

Otherwise looks great.

**Comment by bryangaledxw** [comment]
@selina-dxw The fields we get for a person are "sex", which will always be present, and "gender identity" which may or may not be present - at the moment it'll always show "sex", and will show "gender identity" if present. Should I hide sex if we have gender?

There's an upcoming ticket for displaying "can share". I think we had a conversation before about showing "accommodation required from" - that it was worth having here, as the date that gets prefilled on the booking form might differ from this, if the user manually puts a different date in the bedspace search

**Comment by selina-dxw** [comment]
Oh yep I remember about 'accommodation required from' now, sorry. 

Yes please I think we should hide 'sex' if 'gender identity' is available.

**Pull request merged**